File: clients/ruby/spec/features/call_spec.rb
# frozen_string_literal: true

require_relative "support/context/acceptance"

RSpec.describe "ruby client api: call" do
  include_context "acceptance"

  it "calls a proc" do
    expect(client.call("core.ping")).to be(true)
  end

  it "calls a proc with input" do
    expect(client.call("core.echo", "foo")).to eq("foo")
  end

  it "calls a proc with input and arguments" do
    expect(client.call("type.string.truncate", "foo", length: 1)).to eq("f")
  end

  it "calls with composed input" do
    expect(client.call("type.string.truncate", client.core.echo("foo"), length: 1)).to eq("f")
  end

  it "calls with a composed argument" do
    expect(client.call("type.string.truncate", "foo", length: client.core.echo(1))).to eq("f")
  end
end

File: clients/javascript/__tests__/features/call.spec.js
require("../support/acceptance");

const Proc = require("../../index");
const client = Proc.connect(process.env.SECRET);

test("calls a proc", async () => {
  expect(await client.call("core.ping")).toEqual(true);
});

test("calls a proc with input", async () => {
  expect(await client.call("core.echo", "foo")).toEqual("foo");
});

test("calls a proc with input and arguments", async () => {
  expect(await client.call("type.string.truncate", "foo", { length: 1 })).toEqual("f");
});

test("calls a proc with composed input", async () => {
  expect(await client.call("type.string.truncate", client.core.echo("foo"), { length: 1 })).toEqual("f");
});

test("calls a proc with a composed argument", async () => {
  expect(await client.call("type.string.truncate", "foo", { length: client.core.echo(1) })).toEqual("f");
});
